straxen/rucio.py

```diff
@@ -0,0 +1,360 @@
+
+        self.backends = []
+        if local_rse:
+            rucio_prefix = self.get_rse_prefix(local_rse)
+            self.backends.append(RucioLocalBackend(rucio_prefix))
+
+        self.local_rse = local_rse
+
+        if include_remote:
+            self.backends.append(RucioRemoteBackend(staging_dir))
+
+        # find run numbers to consider
+        if runs_to_consider:
+            self.runs_to_consider = runs_to_consider
+        else:
+            if minimum_run_number is None:
+                minimum_run_number = 0
+            query = {'number': {'$gte': minimum_run_number}}
+            self.runs_to_consider = [r['number'] for r in self.collection.find(query, {'number': 1})]
+
+    def _scan_runs(self, store_fields):
+        if self.local_rse is None:
+            # Don't allow this to be called since it will just loop over all data.
+            raise TooMuchDataError("We don't want use the _scan_runs for the remote rucio backed "
+                                   "as it will just return all our data. ")
+        else:
+            if self.local_did_cache is None:
+                datasets = self.get_rse_datasets(self.local_rse)
+                self.local_did_cache = datasets
+
+            # sometimes there's crap in rucio from testing days, so lets do a query to find the real TPC data there
+            # plus we will have MC data in rucio as well at some point
+            query = {'data.did': {'$in': datasets}}
+            projection = {field: 1 for field in store_fields}
+            # don't care about the _id
+            projection['_id'] = 0
+
+            cursor = self.collection.find(query, projection=projection)
+            for doc in cursor:
+                yield doc
+
+    def find_several(self, keys, **kwargs):
+        if kwargs['fuzzy_for'] or kwargs['fuzzy_for_options']:
+            raise NotImplementedError("Can't do fuzzy with RunDB yet.")
+        if not len(keys):
+            return []
+
+        if self.local_did_cache is None:
+            datasets = self.get_rse_datasets(self.local_rse)
+            self.local_did_cache = datasets
+
+        ret = []
+        for key in keys:
+            did = self.key_to_rucio_did(key)
+            backend_key = f'{key.run_id}-{key.data_type}-{key.lineage_hash}'
+            if did in self.local_did_cache and self.did_is_local(did):
+                ret.append((strax.rucio.__name__, backend_key))
+            else:
+                ret.append(False)
+        return ret
+
+    def _find(self, key: strax.DataKey, write, allow_incomplete, fuzzy_for, fuzzy_for_options):
+        if fuzzy_for or fuzzy_for_options:
+            raise NotImplementedError("Can't do fuzzy with rucio yet.")
+
+        did = self.key_to_rucio_did(key)
+        if self.did_is_local(did):
+            backend_key = f'{key.run_id}-{key.data_type}-{key.lineage_hash}'
+            return "RucioLocalBackend", backend_key
+        else:
+            return "RucioRemoteBackend", did
```

Comment:
We should check that this backend is registered. For example I hit this line (without remote) for `017634-peaklets`

Reply:
Done, thanks!